### PR TITLE
Implement universal line reader on top of asyncio StreamReader

### LINKED ISSUE #74: Separator is not found, and chunk exceed the limit
I am trying to transcode using `django-video-transcoding` which uses fffw library, and for a few streams I am getting `Separator is not found, and chunk exceed the limit`. I tried running ffmpeg with the exact command which`django-video-transcoding` logs and it works fine.

### PR DESCRIPTION
**Pull request by tumb1er**
This allows to read lines not only with \n separator but also with \r\n and \r.

Closes #74

### CODE REVIEW

File: fffw/wrapper/base.py [hunk 1]
```diff
@@ -5,13 +5,82 @@
...
+            # empty read means that stream is closed
+            self.at_eof = len(block) == 0
+
+            # checking max buffer size
+            buffered = len(self.buffer) + len(block)
+            if buffered > self.bufsize:
+                raise asyncio.LimitOverrunError("buffer overrun", buffered)
+
+            self.buffer += block
+            if self.buffer:
```
Comment (ttyS15) [inline fffw/wrapper/base.py:63]: I guess that only around one chance in a million False will be here.

File: fffw/wrapper/base.py [hunk 2]
```diff
@@ -5,13 +5,82 @@
...
+                for line in self.drain_buffer():
+                    yield line
```
Comment (ttyS15) [inline fffw/wrapper/base.py:65]: My be better to replace the 3 lines above with 

```
        [*lines, self.buffer] = self.buffer.splitlines(keepends=True)
        for line in lines:
            yield line.decode(self.encoding)
```

**Review by ttyS15 — APPROVED** [review]

File: fffw/wrapper/base.py (hunk 1 above)
Comment (tumb1er) [reply fffw/wrapper/base.py:63]: It is covered by a test with zero bytes in stdout

File: fffw/wrapper/base.py (hunk 2 above)
Comment (tumb1er) [reply fffw/wrapper/base.py:65]: Do you want to remove `drain_buffer` method completely? It handles a special case with `\r\n`:
```
read_block_1\r
\nread_block_2
```

Comment (ttyS15) [reply fffw/wrapper/base.py:65]: I offer to insert the logic of `drain_buffer` right here.

File: fffw/wrapper/base.py (hunk 1 above)
Comment (ttyS15) [reply fffw/wrapper/base.py:63]: Ok, but I just suppose that the code below will work properly without checking `self.buffer`.

Comment (tumb1er) [reply fffw/wrapper/base.py:63]: I guess it will yield extra empty line when it should not.

File: fffw/wrapper/base.py (hunk 2 above)
Comment (tumb1er) [reply fffw/wrapper/base.py:65]: ok

**Pull request merged**